telegram/ext/dispatcher.py

```diff
@@ -229,30 +236,84 @@ def _pooled(self):
                 break
 
             promise.run()
+
             if isinstance(promise.exception, DispatcherHandlerStop):
                 self.logger.warning(
                     'DispatcherHandlerStop is not supported with async functions; func: %s',
                     promise.pooled_function.__name__)
+            elif isinstance(promise.exception, Exception):
+                if promise.pooled_function in self.error_handlers:
+                    self.logger.error('An error was raised while processing the update '
+                                      'and an uncaught error was raised while handling '
+                                      'the error with an error_handler')
+                elif promise.error_handler:
```

Comment:
> 1. Why is there a `promise.error_handler`: when is it used and why?

`run_async` can not only be used for running handler callbacks asyncronously, but also custom functions. Custom functions will usually have a other arguments than `update, context`, so it might make sense to have special error handlers form them. So the use case is

```python
def custom_func(a, b, c=None):
   raise RuntimeError()

def error_handler(update, context, a, b, c=None):
   print(f'The error {context.error} happend while running custom_func with the arguments {a}, {b}, {c} for update {update}')

def my_callback(update, context):
    # do some stuff
    context.dispatcher.run_async(custom_func, a, b, update=update, error_handler=error_handler, c=c)
```

> 2. What's the difference between having `promise.update` and not having one?

I was gonna argue about distinguishing handler callbacks from custom functions but acutally I see that it's not really straightforward. I think it would be good to restructure and I have two options in mind:

1. Still allow to set `error_handler`, which can be helpful for custom functions. But the signature can just always be 
    ```python
    def error_handler(update, context, *args, **kwargs):
      ...
    ```
    in case `update` is not passed to `run_async`, it will just be `None` (as is already the case for the usual error handlers called on errors happening in jobs)
    If no `error_handler` is passed, use the dispatchers usual error handlers with the downside of not having access to the parameters of the function that raised the error
2. Remove `error_handler` all together and just use the usual error_handlers. To allow acces to the pooled functions parameters, we could add them to the `context` argument (`context.error_(kw)args`). Maybe even the complete promise (`context.promise`)

Looking at the changes with a bit of distance, I'd acutally go for 2.